ext/ImplicitDifferentiationChainRulesExt.jl

```diff
@@ -1,21 +1,23 @@
 module ImplicitDifferentiationChainRulesExt
 
 using AbstractDifferentiation: ReverseRuleConfigBackend, pullback_function
-using ChainRulesCore: ChainRulesCore, NoTangent, RuleConfig, ZeroTangent, unthunk
-using ImplicitDifferentiation: ImplicitFunction, PullbackMul!, check_solution
+using ChainRulesCore: ChainRulesCore, NoTangent, RuleConfig, ZeroTangent, rrule, unthunk
+using ImplicitDifferentiation:
+    ImplicitFunction, PullbackMul!, ReturnByproduct, presolve, solve
+using LinearAlgebra: lmul!, mul!
 using LinearOperators: LinearOperator
 using SimpleUnPack: @unpack
 
 """
     rrule(rc, implicit, x; kwargs...)
-    rrule(rc, implicit, x, Val(return_byproduct); kwargs...)
+    rrule(rc, implicit, x, ReturnByproduct(); kwargs...)
 
 Custom reverse rule for an [`ImplicitFunction`](@ref), to ensure compatibility with reverse mode autodiff.
 
-This is only available if ChainRulesCore.jl is loaded (extension).
+This is only available if ChainRulesCore.jl is loaded (extension), except on Julia < 1.9 where it is always available.
 
-- If `return_byproduct=false` (the default), this returns a single output `y(x)` with a pullback accepting a single cotangent `̄y`.
-- If `return_byproduct=true`, this returns a couple of outputs `(y(x),z(x))` with a pullback accepting a couple of cotangents `(̄y, ̄z)` (remember that `z(x)` is not differentiated so its cotangent is ignored).
+- By default, this returns a single output `y(x)` with a pullback accepting a single cotangent `dy`.
+- If `ReturnByproduct()` is passed as an argument, this returns a couple of outputs `(y(x),z(x))` with a pullback accepting a couple of cotangents `(dy, dz)` (remember that `z(x)` is not differentiated so its cotangent is ignored).
 
 We compute the vector-Jacobian product `Jᵀv` by solving `Aᵀu = v` and setting `Jᵀv = -Bᵀu`.
```

Comment:
`A` and `B` are introduced for the first time here?

Reply:
Yeah my bad ill reshuffle

Reply:
Actually no they're introduced in the main docstring, this is an extension method so it's fine

Reply:
Added a link